app/Api/V1/Domain/Problem/Repository/ProblemRepository.php

```diff
@@ -3,11 +3,26 @@
 namespace App\Api\V1\Domain\Problem\Repository;
 
 use App\Api\V1\Domain\Problem\Entity\Problem;
+use Illuminate\Contracts\Pagination\Paginator;
 
 class ProblemRepository
 {
     public function createOne(array $data): Problem
     {
         return Problem::create($data);
     }
+
+    public function readMany(?int $ownerId, $limit): Paginator
+    {
+        if ($ownerId) {
```

Comment:
This doesn't look good
I think you should create BuildQuery function with queryParam input and return array of where condition